springfox-swagger2/src/main/java/springfox/documentation/swagger2/web/HostNameProvider.java

```diff
@@ -31,8 +32,8 @@ public HostNameProvider() {
     throw new UnsupportedOperationException();
   }
 
-  static UriComponents componentsFrom(HttpServletRequest request) {
-    ServletUriComponentsBuilder builder = ServletUriComponentsBuilder.fromServletMapping(request);
+  static UriComponents componentsFrom(final HttpServletRequest request, final String basePath) {
```

Comment:
I didn't fix this complexity issue since I don't think I changed the complexity of this method.  Let me know if you want me to fix this.

Although, I'm a little confused as to how it ends up a complexity of 800.  My understanding is that the complexity number is calculated based  on the number of paths through a method and I cannot figure out how there are 800 paths through that method.